src/routes/userRoutes.test.js

```diff
@@ -3,36 +3,41 @@ jest.mock('../../lib/users')
 jest.mock('../../sequelize/db')
 
 const db = require('../../sequelize/db')
-const {resetPassword} = require('./userRoutes')
-const {sendPasswordResetEmail} = require('../../lib/users')
-
-const mockJson = jest.fn()
+const {resetPassword, createUser, deleteUser} = require('./userRoutes')
+const {sendPasswordResetEmail, signUp} = require('../../lib/users')
+
+const mcokFindOne = jest.fn()
+db.getModels = () => {
+  return {
+    Accounts: {
+      findOne: mcokFindOne
+    }
+  }
+}
 
 const res = {
-  status: jest.fn().mockReturnValue({json: mockJson})
+  status: jest.fn().mockImplementation(() => {
+    return res
+  }),
+  json: jest.fn()
 }
```

Comment:
i actually changed my code  
`status: jest.fn().mockImplementation(() => {
     return res
   }),
   json: jest.fn()` to 
`status: jest.fn().mockReturnValue({json: mockJson}) ` 😞

Reply:
we shall make an agreement to which way to choose

Reply:
We want `status` to be mocked as well. Because we want to see what argument is being passed to `status` function. Does it make sense?

Reply:
agree that we needs to check both status code as well as the json-ed message

Reply:
> We want `status` to be mocked as well. Because we want to see what argument is being passed to `status` function. Does it make sense?

This was a stupid comment. I didn't read your comment properly at that moment. Your argument is valid. Both way work well. I like mine because I think it's the way how actuall `app.status` is. We can do `app.json()` and `app.status().json()` which means `app.staus` returns `app` itself.

Reply:
this is like choosing between `1+2=3` or `2 + 1 = 3`. Just randomly pick one and stick to it :)